src/timeline/Timeline.js

```diff
@@ -0,0 +1,216 @@
+      _.min(_.map(packedEvents, 'top')) - this.calendarHeight / (end - start);
+    initPosition = initPosition < 0 ? 0 : initPosition;
+    this.state = {
+      _scrollY: initPosition,
+      packedEvents,
+    };
+  }
+
+  componentWillReceiveProps({events, start = 0}) {
+    const width = dimensionWidth - LEFT_MARGIN;
+    this.setState({
+      packedEvents: populateEvents(events, width, start),
+    });
+  }
+
+  componentDidMount() {
+    this.props.scrollToFirst && this.scrollToFirst();
+  }
+
+  scrollToFirst() {
+    setTimeout(() => {
+      if (this.state && this.state._scrollY && this._scrollView) {
+        this._scrollView.scrollTo({
+          x: 0,
+          y: this.state._scrollY,
+          animated: true,
+        });
+      }
+    }, 1);
+  }
+
+  // _renderRedLine() {
+  //   const offset = 100;
+  //   const {start = 0} = this.props;
+  //   const timeNowHour = moment().hour();
+  //   const timeNowMin = moment().minutes();
+  //   return (
+  //     <View
+  //       key={'timeNow'}
+  //       style={[
+  //         this.styles.lineNow,
+  //         {
+  //           top: offset * (timeNowHour - start) + (offset * timeNowMin) / 60,
+  //           width: dimensionWidth - 20,
+  //         },
+  //       ]}
+  //     />
+  //   );
+  // }
+
+  _renderLines() {
+    const {format24h, start = 0, end = 24} = this.props;
+    const offset = this.calendarHeight / (end - start);
+
+    return range(start, end + 1).map((i, index) => {
+      let timeText;
+      if (i === start) {
+        timeText = '';
+      } else if (i < 12) {
+        timeText = !format24h ? `${i} AM` : i;
+      } else if (i === 12) {
+        timeText = !format24h ? `${i} PM` : i;
+      } else if (i === 24) {
+        timeText = !format24h ? '12 AM' : 0;
+      } else {
+        timeText = !format24h ? `${i - 12} PM` : i;
+      }
+      return [
+        <Text
+          key={`timeLabel${i}`}
+          style={[this.styles.timeLabel, {top: offset * index - 6}]}>
+          {timeText}
+        </Text>,
+        i === start ? null : (
+          <View
+            key={`line${i}`}
+            style={[
+              this.styles.line,
+              {top: offset * index, width: dimensionWidth - 20},
+            ]}
+          />
+        ),
+        <View
+          key={`lineHalf${i}`}
+          style={[
+            this.styles.line,
+            {top: offset * (index + 0.5), width: dimensionWidth - 20},
+          ]}
+        />,
+      ];
+    });
+  }
+
+  _renderTimeLabels() {
+    const {start = 0, end = 24} = this.props;
+    const offset = this.calendarHeight / (end - start);
+    return range(start, end).map((item, i) => {
+      return (
+        <View key={`line${i}`} style={[this.styles.line, {top: offset * i}]} />
+      );
+    });
+  }
+
+  _onEventTapped(event) {
+    this.props.eventTapped(event);
+  }
+
+  _renderEvents() {
+    const {packedEvents} = this.state;
+    let events = packedEvents.map((event, i) => {
```

Comment:
When you render your events by order you get a scenario where parallel events are stacked in the wrong order. Meaning the event that start later should cover the earlier one, even if it was first in the list, but it doesn't. You can see that when you press and event and get a peek to the event behind it duo to the TouchableOpacity's active opacity.

Reply:
I didn't get this completely, but i have reduced TouchableOpacity, let me know if that fix this issue.

Reply:
I am not sure if we can resolve this.